cg/store/api/find_business_data.py

```diff
@@ -324,6 +324,57 @@ def get_number_of_reads_for_sample_from_metrics(self, sample_internal_id: str) -
         reads_count: Optional[int] = total_reads_query.scalar()
         return reads_count if reads_count else 0
 
+    def get_samples_on_flow_cell_from_metrics(self, flow_cell_name: str) -> List[Sample]:
+        """Get samples on a given flow cell."""
+        metrics = apply_metrics_filter(
+            metrics=self._get_query(table=SampleLaneSequencingMetrics),
+            filter_functions=[SequencingMetricsFilter.FILTER_BY_FLOW_CELL_NAME],
+            flow_cell_name=flow_cell_name,
+        ).all()
+
+        return [
+            self.get_sample_by_internal_id(internal_id=metric.sample_internal_id)
+            for metric in metrics
+        ]
+
+    def get_average_passing_q30_for_sample_from_metrics(self, sample_internal_id: str) -> float:
+        """Get average q30 for a sample on a given flow cell."""
+        metrics = apply_metrics_filter(
+            metrics=self._get_query(table=SampleLaneSequencingMetrics),
+            filter_functions=[SequencingMetricsFilter.FILTER_BY_SAMPLE_INTERNAL_ID],
+            sample_internal_id=sample_internal_id,
+        ).all()
+        total_passing_q30: float = 0
+        for metric in metrics:
+            total_passing_q30 += metric.sample_base_fraction_passing_q30
+
+        return total_passing_q30 / len(metrics) if total_passing_q30 else 0
+
+    def get_number_of_reads_for_flow_cell(self, flow_cell_name: str) -> int:
+        """Get number of reads for flow cell from sample lane sequencing metrics."""
+        samples: List[Sample] = self.get_samples_on_flow_cell_from_metrics(
+            flow_cell_name=flow_cell_name
+        )
+        reads_count: int = 0
+        for sample in samples:
+            reads_count += self.get_number_of_reads_for_sample_from_metrics(
+                sample_internal_id=sample.internal_id
+            )
+        return reads_count if reads_count else 0
+
+    def get_average_passing_q30_for_samples_on_flow_cell(self, flow_cell_name: str) -> float:
+        """Get average q30 for samples on a given flow cell."""
+        samples: List[Sample] = self.get_samples_on_flow_cell_from_metrics(
+            flow_cell_name=flow_cell_name
+        )
+        total_passing_q30: int = 0
+        for sample in samples:
+            total_passing_q30 += self.get_average_passing_q30_for_sample_from_metrics(
+                sample_internal_id=sample.internal_id
+            )
+
+        return total_passing_q30 / len(samples) if total_passing_q30 else 0
```

Comment:
This probably won't be correct, right? I think you need to do a new filter function that filters for both `sample.internal_id` and `flowcell.name`. If you do this on a sample that has been on multiple flow cells you would return the average over all flow cells.

Reply:
you are very right with this!